src/mlpack/methods/logistic_regression/logistic_regression_function.hpp

```diff
@@ -107,6 +127,23 @@ class LogisticRegressionFunction
                 const size_t i,
                 GradType& gradient) const;
 
+  /**
+   * Evaluate the gradient of the logistic regression log-likelihood function
+   * with the given parameters, for the given batch size from a given point the
+   * in dataset. This is useful for optimizers such as SGD, which require a
+   * separable objective function.
+   *
+   * @param parameters Vector of logistic regression parameters.
+   * @param begin Index of the starting point to use for objective function gradient evaluation.
+   * @param batchSize Number of points to be processed as a batch for objective function gradient evaluation.
+   * @param gradient Vector to output gradient into.
+   */
+
+  void Gradient(const arma::mat& parameters,
+                const size_t begin,
+                const size_t batchSize,
+                arma::mat& gradient) const;
+
```

Comment:
I think we need to also add a `Shuffle()` method to this class.

Reply:
I think we still need an implementation here, and I think the NCA error function also needs a `Shuffle()` implementation.  I think it might be better to modify the ordering of the data points in these cases instead of holding a `visitationOrder` structure.